src/Database/Expression/QueryExpression.php

```diff
@@ -439,6 +439,18 @@ public function count()
     }
 
     /**
+     * Builds equal condition or assignment with identifier wrapping.
+     *
+     * @param string $left Left join condition field name.
+     * @param string $right Right join condition field name.
+     * @return $this
+     */
+    public function equalFields($left, $right)
+    {
+        return $this->eq(new IdentifierExpression($left), new IdentifierExpression($right));
```

Comment:
I think we we should test for $left and $right not being `ExpressionInterface` before wrapping them in an identifier expression